htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxReswap.java

```diff
@@ -0,0 +1,87 @@
+package io.github.wimdeblauwe.htmx.spring.boot.mvc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.core.annotation.AliasFor;
+
+/**
+ * Annotation to specify how the response will be swapped.
+ * See <a href="https://htmx.org/attributes/hx-swap/">hx-swap</a> for possible values.
+ *
+ * @see <a href="https://htmx.org/reference/#response_headers">HX-Reswap</a>
+ */
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface HxReswap {
+
+    /**
+     * A value to specify how the response will be swapped.
+     *
+     * @see <a href="https://htmx.org/attributes/hx-swap/">hx-swap</a>
+     */
+    HxSwapType value() default HxSwapType.INNER_HTML;
+
+    /**
+     * Set the time in milliseconds that should elapse after receiving a response to swap the content.
+     */
+    long swap() default -1;
```

Comment:
I am wondering if we should use `Duration` here?

Reply:
You can't use Duration here. See [9.6.1. Annotation Interface Elements](https://docs.oracle.com/javase/specs/jls/se17/html/jls-9.html#jls-9.6.1)

>The return type of a method declared in the body of annotation interface must be one of the following, or a compile-time error occurs:
>
>- A primitive type
>- String
>- Class or an invocation of Class ([§4.5](https://docs.oracle.com/javase/specs/jls/se17/html/jls-4.html#jls-4.5))
>- An enum class type
>- An annotation interface type
>- An array type whose component type is one of the preceding types ([§10.1](https://docs.oracle.com/javase/specs/jls/se17/html/jls-10.html#jls-10.1)).

Reply:
Ah, ok, that is a pity.